kratos/input_output/vtk_output.cpp

```diff
@@ -0,0 +1,452 @@
+
+    // write elements
+    for (const auto& elem : local_mesh.Elements())
+    {
+        const auto& elem_geometry = elem.GetGeometry();
+        const unsigned int number_of_nodes = elem_geometry.size();
+
+        WriteScalarDataToFile((unsigned int)number_of_nodes, rFileStream);
+        for (const auto& node : elem_geometry){
+            if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << " ";
+            int id = mKratosIdToVtkId[node.Id()];
+            WriteScalarDataToFile((int)id, rFileStream);
+        }
+        if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << "\n";
+    }
+
+    // write Conditions
+    for (const auto& condition : local_mesh.Conditions())
+    {
+        const auto& condition_geometry = condition.GetGeometry();
+        const unsigned int number_of_nodes = condition_geometry.size();
+
+        WriteScalarDataToFile((unsigned int)number_of_nodes, rFileStream);
+        for (const auto& node : condition_geometry){
+            if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << " ";
+            int id = mKratosIdToVtkId[node.Id()];
+            WriteScalarDataToFile((int)id, rFileStream);
+        }
+        if (mFileFormat == VtkOutput::FileFormat::VTK_ASCII) rFileStream << "\n";
+    }
+
+}
+
+void VtkOutput::WriteConditionAndElementTypes(const ModelPart &rModelPart, std::ofstream& rFileStream)
+{
+    const auto& local_mesh = rModelPart.GetCommunicator().LocalMesh();
+    // write cell types header
+    rFileStream << "\nCELL_TYPES " << local_mesh.NumberOfConditions() + local_mesh.NumberOfElements() << "\n";
+
+    // write elements types
+    for (const auto& elem : local_mesh.Elements())
+    {
+        int  cell_type = -1;
+
+        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Triangle2D3)
+             cell_type = 5; // triangle 2d
+        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Quadrilateral2D4)
+             cell_type = 9; // quad 2d
+        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Tetrahedra3D4)
+             cell_type = 10; // tetra 3d
+        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Line2D2)
+             cell_type = 3; // line 2d or 3d
+        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Line3D2)
+             cell_type = 3; // line 2d or 3d
+        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Point2D)
+             cell_type = 1;
+        if (elem.GetGeometry().GetGeometryType() == GeometryData::KratosGeometryType::Kratos_Point3D)
+             cell_type = 1;
+        if( cell_type < 0)
+            KRATOS_ERROR<<"Modelpart contains elements with geometries for which no VTK-output is implemented!"<<std::endl;
```

Comment:
suggestion instead of those two long loops:
~~~c
std::unordered_map<GeometryData::KratosGeometryType, int> {
    {Kratos_Triangle2D3, 5 },
   etc
}
~~~

then you can only call the map to get the cell-type and throw if it returns `end`